test/api/settings_test.rb

```diff
@@ -0,0 +1,68 @@
+require 'test_helper'
+require 'json'
+
+class SettingTest < ActiveSupport::TestCase
+    include Rack::Test::Methods
+    include TestHelpers::AuthHelper
+    include TestHelpers::JsonHelper
+
+    def app
+        Rails.application
+    end
+
+    # Get config details
+    def test_get_config_details        
+        expected_ProductName =  Doubtfire::Application.config.institution[:product_name]
+        
+        # Perform the GET
+        get '/api/settings'
+
+        # Set returned details
+        returned_mes = last_response_body['externalName']
+
+        # Check if the call succeeds
+        assert_equal 200, last_response.status
+        # Check returned details match as expected 
+        assert_equal expected_ProductName, returned_mes
+    end     
+
+    # Get privacy policy details
+    def test_get_privacy_policy_details
+        expected_privacy = Doubtfire::Application.config.institution[:privacy]
+        expected_plagiarism = Doubtfire::Application.config.institution[:plagiarism]
+        
+        # Perform the GET
+        get '/api/settings/privacy'
+        
+        # Set two returned details
+        returned_privacy = last_response_body['privacy']
+        returned_plagiarism = last_response_body['plagiarism']
+
+        # Check if the call succeeds
+        assert_equal 200, last_response.status
+
+        # Check returned details match as expected
+        assert_equal expected_privacy, returned_privacy
+        assert_equal expected_plagiarism, returned_plagiarism
+    end
+
+    # Get all unit teaching periods info
+    def test_get_all_unit_teaching_periods_info
+        # The GET we are testing
+        get '/api/teaching_periods/id/units'
```

Comment:
Do we need this API endpoint? It does not sound useful to me. Can you check where it is used in the front end and recommend we delete it if it is not used.

Reply:
I have checked the api endpoint, and as you said we don't need it. 
So do I need to delete this link (get '/ api / teaching_periods / id / units') and its test method (def test_get_all_unit_teaching_periods_info) and commit setting_test.rb again?

Reply:
Sorry, I thought I had replied to this earlier. Yes, please remove the test and the end point.

Reply:
Once that is done I can merge the PR.